lib/database-ops.js

```diff
@@ -101,46 +101,82 @@ var changeUserPasswordInDatabase = function(userID, hashedPassword) {
 
 var startDatabaseClient = function(callback, options) {
     var dbURL = (options !== undefined && options.dbURL !== undefined) ? options.dbURL : url;
-    MongoClient.connect(dbURL, function(err, db) {
-        if (err) {
-            callback(err);
-            return;
-        }
+    MongoClient.connect(dbURL).then((db) => {
         logger.info("Successfully connected to database.");
         database = db;
 
-        var finishInitialization = function () {
-            imageEntryCollection = database.collection("image-entries");
+        var finishInitialization = function () {  
             commentsCollection = database.collection("comments");
             logCollection = database.collection("logs");
             actionHistoryCollection = database.collection("action-history");
 
             usersCollection.createIndex({ username: 1 }, { unique: true, sparse: true });
-            callback(err);
+            callback();
         };
 
-        database.listCollections().toArray(function(err, collections) {
-            if (err) {
-                throw err;
+        var createIndexForImageEntriesCollection = function(collection) {
+            collection.createIndex({ uploadeddate: 1 }, { 
+                name: "expire_index",
+                expireAfterSeconds: parseInt(process.env.EXPIRE_AFTER_SECONDS) || 300,
+                partialFilterExpression: {
+                    temp: {
+                        $eq: true,
+                    }
+                } 
+            }).then(() => {
+                logger.info("expire_index created for images collection.");
+            });
+        }
+
+        var checkForImageEntriesCollection = function(collectionNames) {
+            if(collectionNames.indexOf("image-entries") < 0) {
+                database.createCollection("image-entries").then((collection) => {
+                    createIndexForImageEntriesCollection(collection);
+                    imageEntryCollection = collection;
+                    logger.info("Images collection created.");
```

Comment:
```suggestion
                    imageEntryCollection = collection;
                    logger.info("Images collection created.");
                    createIndexForImageEntriesCollection(collection);
```
Just reordering the log statement and setting the variable for image collection to be before creating the index. I'd be surprised if the index created log appears before the image collection created log because index creation is async, but just thought I'd move it up for sake of clarity.

Reply:
Yes, it would be misleading :sweat_smile:.